cpp/src/arrow/array/builder_nested.h

```diff
@@ -215,13 +216,14 @@ class ARROW_EXPORT LargeListBuilder : public BaseListBuilder<LargeListType> {
 /// Key uniqueness and ordering are not validated.
 class ARROW_EXPORT MapBuilder : public ArrayBuilder {
  public:
-  /// Use this constructor to incrementally build the key and item arrays along with
-  /// offsets and null bitmap.
+  /// Use this constructor to define the built array's type explicitly. If key_builder or
+  /// item_builder has indeterminate type, this builder will also.
```

Comment:
I'm not sure I understand the comment. I can pass an explicit type but the type may still be indeterminate?